src/Conversion/ONNXToKrnl/Sequence/SequenceErase.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------SequenceErase.cpp - Lowering SequenceErase Op-----------------=== //
+//
+// Copyright 2020-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX SequenceErase Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+#include "src/Dialect/Krnl/KrnlHelper.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+
+using namespace mlir;
+
+struct ONNXSequenceEraseOpLowering : public ConversionPattern {
+  ONNXSequenceEraseOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(typeConverter,
+            mlir::ONNXSequenceEraseOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    Location loc = op->getLoc();
+    ONNXSequenceEraseOpAdaptor operandAdaptor(operands);
+    ONNXSequenceInsertOp thisOp = dyn_cast<ONNXSequenceInsertOp>(op);
+    MultiDialectBuilder<KrnlBuilder, MathBuilder> create(rewriter, loc);
```

Comment:
Please add also a `MemRefBuilder` so that later you can use it for memref dialect operations.